### PR TITLE
Use random secret for logs channel

### PR DESCRIPTION
**Pull request by pcarranzav**
And fix reporting of new API secret upon regeneration.

### CODE REVIEW

File: src/utils.coffee [hunk 1]
```diff
@@ -102,27 +102,28 @@ exports.connectivityCheck = _.once ->
 				blink.pattern.start(networkPattern)
 
 
-apiSecretPromise = null
-generateApiSecret = ->
+secretPromises = {}
+generateSecret = (name) ->
 	Promise.try ->
-		return config.forceApiSecret ? randomHexString.generate()
+		return config.forceApiSecret if name == 'apiSecret' && config.forceApiSecret?
```
Comment (petrosagg) [inline src/utils.coffee]: You made the function generic, but this if in here feels dirty. A separate specific function for `'apiSecret'` a way to resolve this.

File: src/utils.coffee [hunk 2]
```diff
@@ -102,27 +102,28 @@ exports.connectivityCheck = _.once ->
...
+		knex('config').update(secretInDB).where(key: name)
 		.then (affectedRows) ->
 			knex('config').insert(secretInDB) if affectedRows == 0
 		.return(newSecret)
 
-exports.newApiSecret = newApiSecret = ->
-	apiSecretPromise ?= Promise.resolve()
-	apiSecretPromise = apiSecretPromise.then ->
-		generateApiSecret()
+exports.newSecret = newSecret = (name) ->
```
Comment (Page-) [inline src/utils.coffee]: This local `newSecret` define doesn't seem to be used anywhere?

File: src/utils.coffee (hunk 1 above)
Comment (Page-) [inline src/utils.coffee]: Alternatively you could have
```coffee
config.forceSecret =
	apiSecret: '...'
	other: '...'

return config.forceSecret[name] if config.forceSecret[name]?
```

File: src/utils.coffee [hunk 3]
```diff
@@ -131,7 +132,7 @@ exports.extendEnvVars = (env, uuid) ->
 		RESIN_SUPERVISOR_ADDRESS: "http://#{host}:#{config.listenPort}"
 		RESIN_SUPERVISOR_HOST: host
 		RESIN_SUPERVISOR_PORT: config.listenPort
-		RESIN_SUPERVISOR_API_KEY: exports.getOrGenerateApiSecret()
+		RESIN_SUPERVISOR_API_KEY: exports.getOrGenerateSecret('apiSecret')
```
Comment (Page-) [inline src/utils.coffee]: This would make more sense as just `'api'`, since the fact that it's a secret is already clear from the function name

Comment (pcarranzav) [inline src/utils.coffee]: The thing is the same string is also used as a key to store the secret in the DB, and I'd rather keep it explicit there.

Comment (Page-) [inline src/utils.coffee]: `knex('config').update(secretInDB).where(key: "#{name}Secret")`?

Comment (pcarranzav) [inline src/utils.coffee]: Fair enough :grin:

Comment (pcarranzav) [inline src/utils.coffee]: @Page- done.

File: src/api.coffee [hunk 4]
```diff
@@ -129,9 +129,9 @@ module.exports = (application) ->
 	# Expires the supervisor's API key and generates a new one.
 	# It also communicates the new key to the Resin API.
 	api.post '/v1/regenerate-api-key', (req, res) ->
-		utils.newApiSecret()
+		utils.newSecret('api')
 		.then (secret) ->
-			device.updateState(apikey: secret)
+			device.updateState(api_secret: secret)
```
Comment (Page-) [inline src/api.coffee]: Why did you change this property name? It needs to match up with the api

Comment (Page-) [inline src/api.coffee]: Hmm, I guess it's been changed so that it now matches up

**Comment by Page-** [comment]
LGTM

File: src/api.coffee (hunk 4 above)
Comment (pcarranzav) [inline src/api.coffee]: Yeah, in app.coffee we always had it as api_secret, I messed up when I added this endpoint
